go-controller/pkg/ovn/egressip.go

```diff
@@ -791,6 +791,9 @@ func (oc *Controller) addPodEgressIPAssignments(name string, statusAssignments [
 	existing, exists := oc.eIPC.podAssignment[podKey]
 	if !exists {
 		remainingAssignments = statusAssignments
+		if err := oc.eIPC.deletePerPodGRSNAT(pod); err != nil && !errors.Is(err, skippedPodError) {
```

Comment:
> Why would we need to delete this here when if !exists?

`oc.eIPC.podAssignment` tracks what egress IP status has been setup for each pod which matches an egress IP. If nothing exists for a pod, it means nothing has been performed yet and we are going to now. Hence, we need to delete the pod GR SNAT and set up the egress IP SNAT. 

> Why not just always remove an invalid pod GR SNAT or update it during egress IP add?

Because this function will end up getting called from `reconcileEgressIP`, `reconcileEgressIPNamespace`, `reconcileEgressIPPod` for each pod and for each pod update. On restart we might end up here in the order of tens for every single pod, which, even though things should be idempotent, is a bit excessive on the CPU when there is no need to do so.